streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -92,56 +94,58 @@ public void run() {
      * @param config                the {@link StreamsConfig} specified by the user
      * @param metrics               the {@link StreamsMetrics} created by the thread
      * @param stateDirectory        the {@link StateDirectory} created by the thread
-     * @param recordCollector       the instance of {@link RecordCollector} used to produce records
+     * @param producer              the instance of {@link Producer} used to write records
      */
-    public StreamTask(TaskId id,
-                      String applicationId,
-                      Collection<TopicPartition> partitions,
-                      ProcessorTopology topology,
-                      Consumer<byte[], byte[]> consumer,
+    public StreamTask(final TaskId id,
+                      final String applicationId,
+                      final Collection<TopicPartition> partitions,
+                      final ProcessorTopology topology,
+                      final Consumer<byte[], byte[]> consumer,
                       final ChangelogReader changelogReader,
-                      StreamsConfig config,
-                      StreamsMetrics metrics,
-                      StateDirectory stateDirectory,
-                      ThreadCache cache,
-                      Time time,
-                      final RecordCollector recordCollector) {
+                      final Producer<byte[], byte[]> producer,
```

Comment:
It isn't clear to my why we need the `producer` it is used only to pass to the `RecordCollectorImp` that was a ctor arg previously, and to call `close` on. Why can't we just leave it as it was and call `close` on the `RecordCollector`?